doc/nrf/gs_recommended_versions.rst

```diff
@@ -196,8 +196,6 @@ Building documentation
      - |sphinxcontrib-mscgen_ver|
    * - breathe
      - |breathe_ver|
-   * - docutils
```

Comment:
In case of Python dependencies, users should rely on a requirements file, having such a list doesn't really help IMO:

- It's duplicate information (requirements.txt and here)
- It's easy to have information/versions mismatch
- The list includes "subdependencies" (e.g. sphinxcontrib-svg2pdfconverter from Zephyr, which is in fact not needed for NCS as pdf build is not used...). In a few words: hard to maintain, easily out of date.